### PR TITLE
Support mozilla-mobile chains

### PR DESCRIPTION
**Pull request by JohanLorenzo**
~~**Do not merge yet**~~ Ready to land!

### CODE REVIEW

File: scriptworker/constants.py [hunk 1]
```diff
@@ -179,6 +179,7 @@
         "gecko-2-decision",
         "gecko-3-decision",
         "gecko-decision",  # legacy
+        "gecko-focus",  # mozilla-mobile releases
```
Comment (escapewindow) [inline scriptworker/constants.py]: This seems like a misnomer - Focus is not Gecko, correct?
It doesn't block anything, but we may want to spin up a 2nd workerType at some point with a more accurate name, and then retire gecko-focus.

File: scriptworker/cot/verify.py [hunk 2]
```diff
@@ -803,13 +803,13 @@ def get_all_artifacts_per_task_id(chain, upstream_artifacts):
...
             )
         # Download actions.json for decision+action task cot verification
-        if link.task_type in DECISION_TASK_TYPES:
-            add_enumerable_item_to_dict(
-                dict_=all_artifacts_per_task_id, key=link.task_id, item='public/actions.json'
-            )
-            add_enumerable_item_to_dict(
-                dict_=all_artifacts_per_task_id, key=link.task_id, item='public/parameters.yml'
-            )
+        # if link.task_type in DECISION_TASK_TYPES:
```
Comment (escapewindow) [inline scriptworker/cot/verify.py]: For now, we could either create a size 0 file in the Focus decision task, or have a map of `cot_product` to `decision_task_artifacts`. When taskgraph is a generic module, we should get these files for free.

File: scriptworker/cot/verify.py [hunk 3]
```diff
@@ -1395,7 +1395,7 @@ def compare_jsone_task_definition(parent_link, compare_definition):
             mach_commands = ACTION_MACH_COMMANDS
         else:
             mach_commands = DECISION_MACH_COMMANDS
-        verify_decision_command(link, mach_commands=mach_commands)
+        # verify_decision_command(link, mach_commands=mach_commands)
```
Comment (escapewindow) [inline scriptworker/cot/verify.py]: This check should go away when we kill cotv1.

File: scriptworker/cot/verify.py [hunk 4]
```diff
@@ -1637,7 +1637,7 @@ def check_num_tasks(chain, task_count):
         # docker-worker task). So only run these tests if they are not the chain
         # object.
         check_interactive_docker_worker(link)
-        verify_docker_image_sha(chain, link)
+        # verify_docker_image_sha(chain, link)
```
Comment (escapewindow) [inline scriptworker/cot/verify.py]: This check should go away when we kill cotv1.

File: scriptworker/cot/verify.py (hunk 2 above)
Comment (JohanLorenzo) [reply scriptworker/cot/verify.py]: Great. Thanks for the idea!

File: scriptworker/cot/verify.py (hunk 3 above)
Comment (JohanLorenzo) [reply scriptworker/cot/verify.py]: Good idea. I'll do it this way.

File: scriptworker/cot/verify.py [hunk 5]
```diff
@@ -1281,6 +1281,12 @@ def verify_decision_command(decision_link, mach_commands=DECISION_MACH_COMMANDS)
         CoTError: on failure.
 
     """
+    if 'mozilla-mobile/focus-android' in get_source_url(parent_link):
+        log.warn(
+            'Repo "mozilla-mobile/focus-android" does not support JSON-e yet. Skipping parent task verifications'
```
Comment (escapewindow) [inline scriptworker/cot/verify.py]: looks like we're not fully verifying by either cotv1 or cotv2?

Comment (JohanLorenzo) [reply scriptworker/cot/verify.py]: That's right. I think it's okay to leave it until https://github.com/taskcluster/taskcluster-github/pull/239 gets implemented. No APK can be published to users if the decision task gets altered.

Are you okay with it?

Comment (escapewindow) [reply scriptworker/cot/verify.py]: We're not checking the decision task command, and we're not checking the decision task definition - how do we know if it's altered?

Comment (JohanLorenzo) [reply scriptworker/cot/verify.py]: > No APK can be published to users if the decision task gets altered.

My apologies, this wasn't very clear. I meant: if an attackers gets the right scopes and forges a decision task, then this person can't get the APK live (unless the attacker has access to Google Play too). There is no alteration verification per se, like you said.

**Comment by JohanLorenzo** [comment]
I'll add tests about this by-pass, if that's okay to you @escapewindow

File: scriptworker/cot/verify.py (hunk 5 above)
Comment (escapewindow) [reply scriptworker/cot/verify.py]: Ok. As long as the Focus team is aware of this hole and don't publish an apk without double checking that it's one they want to publish, this is probably ok.

File: scriptworker/constants.py [hunk 6]
```diff
@@ -267,6 +277,10 @@
                 'project:comm:thunderbird:releng:signing:cert:nightly-signing': 'all-nightly-branches',
                 'project:comm:thunderbird:releng:signing:cert:release-signing': 'all-release-branches',
             }),
+            'mobile': frozendict({
```
[the remark this answers is not in the document]
Comment (JohanLorenzo) [reply scriptworker/constants.py]: Good point. I think `focus-repo` is better. At the moment, having a single `cot_product` is easier to implement for mobile projects.

File: scriptworker/cot/verify.py (hunk 5 above)
Comment (JohanLorenzo) [reply scriptworker/cot/verify.py]: Actually, having a condition on the `cot_product` instead of the repo is safer. I'll use. Thanks for the hint!

**Comment by JohanLorenzo** [comment]
Rebased on top of master. Still tests to add.

**Comment by escapewindow** [comment]
Ok. Be aware #221 removes a bunch of code --  lmk if you want this to land first.

**Comment by JohanLorenzo** [comment]
I'd like to land it before #211. The PR is now ready, IMO.

**Comment by escapewindow** [comment]
Good work! Do you need any help with shpiping this?

**Comment by JohanLorenzo** [comment]
It's all good. I just deployed it. 

Thanks for the huge help on making focus automatically signed!

**Pull request merged**